src/services/auth.service.ts

```diff
@@ -40,6 +41,13 @@ export class AuthorizationService {
       if (!!tokenIsBlacklisted) {
         return false;
       }
+      if (_roles && _roles.length > 0) {
+        const usersService = Container.get(UsersService);
```

Comment:
This method executes every time a decorator `@Authorized` is called . Maybe you can have this Container.get ( along with the ones to get the JwtService and the RedisService ) on the top of the file or having the dependencies initialized via constructor. What do you think?

Reply:
I agree, this will be more efficient and I think that the code will be cleaner